validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java

```diff
@@ -0,0 +1,114 @@
+package com.amazon.aoc.validators;
+
+import com.amazon.aoc.callers.ICaller;
+import com.amazon.aoc.exception.BaseException;
+import com.amazon.aoc.exception.ExceptionCode;
+import com.amazon.aoc.fileconfigs.FileConfig;
+import com.amazon.aoc.helpers.MustacheHelper;
+import com.amazon.aoc.helpers.RetryHelper;
+import com.amazon.aoc.models.Context;
+import com.amazon.aoc.models.ValidationConfig;
+import com.amazon.aoc.services.CloudWatchService;
+import com.amazonaws.services.logs.model.OutputLogEvent;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.fge.jsonschema.main.JsonSchema;
+import com.github.fge.jsonschema.main.JsonSchemaFactory;
+import com.github.fge.jsonschema.report.ListReportProvider;
+import com.github.fge.jsonschema.report.LogLevel;
+import com.github.fge.jsonschema.report.ProcessingReport;
+import com.github.fge.jsonschema.util.JsonLoader;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class CWLogValidator implements IValidator {
+
+  protected String logStreamName = "otlp-logs";
+
+  private static final String LOGGROUPPATH = "/aws/ecs/otlp/%s/logs";
```

Comment:
nit: This method of building the log group name introduces a strong coupling between the terraform test and the validator. It requires the config to use this exact log group name format. Instead it could be better if the validator is provided the log group and stream name at launch.

Reply:
As in I should add it as an environment variable or have it as a variable established at runtime? Also, looking at other validators, it seems they establish the log group name in a similar fashion. For example for ContainerInsightECSPrometheus Validator, it is established by
`logGroupName =
        String.format(
            "/aws/ecs/containerinsights/%s/%s",
            context.getCloudWatchContext().getClusterName(), "prometheus");`
Does this not have a coupling issue between the tf test and validator?

Reply:
Yes, there is a coupling there also. You don't need to address this comment now.